### DISCUSSION TITLE
Postgres Automatic Recover / Missing Mirth Channels

### DISCUSSION THREAD

**Discussion by rjkroll** [nextgenhealthcare/connect#5371] [Q&A] [answered]
Hi,

Ran into an issue where we were working on a new channel in Mirth last week and it looks like something happened to the power/hardware that causes the Postgres database to perform an automatic recovery from a point in time that was before this channel was created. 

So, the postgres database itself, in mirth_db does not have this channel listed because it "recovered".    Does Mirth, in any folder, keep some sort of copy or data about this channel?  I know it uses Postgres to get its running list of channels it supposed to show, but was hoping to find some miracle solution to find something related to the channel since it is gone from the database.

Thanks,

**Comment by pacmano1** [comment, marked as answer]
It does not.

**Reply by rjkroll** [reply]
Thanks, that's what I thought.    Though, I think we got really, really lucky.     While rooting around in all the Mirth Connect folders I stumbled across the t\appdata\mirthdb\log  directory, which had several .dat files.    I opened these up with NotePad++ and saw that they had XML of the channel we needed written out to it.    Not sure what these files were used for, but maybe they held the last changes of the last channel that we had worked on?    Thankfully, we can see all the Transformer Steps and underlying code, so we can reconstruct our missing channel.     I think we go really, really lucky on this one as we had no backups of anything for this box, which....lesson learned.

Thanks agin.

**Reply by pacmano1** [reply]
Please don't use Derby for anything important, use a real database engine.

**Reply by rjkroll** [reply]
We actually had Mirth pointed to a postgres database for its mirth_db.    I guess these .dat files are related to Derby when we originally installed?   We never actually used a Derby backend or even installed it, unless we installed it by mistake during the install.

**Reply by pacmano1** [reply]
Derby is installed by default.  If you had Derby files in `appdata` the engine at some point was running against Derby.
